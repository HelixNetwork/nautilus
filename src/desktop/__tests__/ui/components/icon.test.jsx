import React from 'react';
import { shallow } from 'enzyme';

import Icon from 'ui/components/icon';

const props = {
    icon: 'hlx',
    size: 32,
    color: '#ff0000',
};

describe('Test for Icon component', () => {
    test('Render the Icon component', () => {
        const wrapper = shallow(<Icon {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Render svg element', () => {
        const wrapper = shallow(<Icon {...props} />);

        expect(wrapper.text()).toEqual('');
    });

    test('Icon size', () => {
        const wrapper = shallow(<Icon {...props} />);

        expect(wrapper.props().style.fontSize).toEqual(32);
        expect(wrapper.props().style.lineHeight).toEqual('32px');
    });

    test('Icon color', () => {
        const wrapper = shallow(<Icon {...props} />);

        expect(wrapper.props().style.color).toEqual('#ff0000');
    });
});
