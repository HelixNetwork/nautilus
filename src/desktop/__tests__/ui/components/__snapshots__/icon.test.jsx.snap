// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Icon component Render the Icon component 1`] = `
<span
  className="icon hlx"
  style={
    Object {
      "color": "#ff0000",
      "fontSize": 32,
      "lineHeight": "32px",
    }
  }
/>
`;
